app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/galleryuplift/GalleryUpliftViewModel.kt

```diff
@@ -0,0 +1,611 @@
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import java.io.FileReader
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.getLastModifiedTime
+import kotlin.io.path.notExists
+
+class GalleryUpliftViewModel(
+    selectorNotEdit: Boolean,
+    val preferences: Preferences,
+    val router: AppRouter,
+    val indexRepo: ResourceIndexRepo,
+    val previewStorageRepo: PreviewProcessorRepo,
+    val metadataStorageRepo: MetadataProcessorRepo,
+    val tagsStorageRepo: TagsStorageRepo,
+    val statsStorageRepo: StatsStorageRepo,
+    val scoreStorageRepo: ScoreStorageRepo,
+    val analytics: GalleryAnalytics,
+) : ViewModel() {
+    private lateinit var index: ResourceIndex
+    private lateinit var tagsStorage: TagStorage
+    private lateinit var previewStorage: PreviewProcessor
+    private lateinit var metadataStorage: MetadataProcessor
+    private lateinit var statsStorage: StatsStorage
+    private lateinit var scoreStorage: ScoreStorage
+    private lateinit var rootAndFav: RootAndFav
+    private lateinit var resourcesIds: List<ResourceId>
+
+    var galleryItems: MutableList<GalleryPresenter.GalleryItem> = mutableListOf()
+    var diffResult: DiffUtil.DiffResult? = null
+
+    private val _selectedResources: MutableList<ResourceId> = mutableListOf()
+    val selectedResources: List<ResourceId> = _selectedResources
+
+    val scoreWidgetController = ScoreWidgetController(
+        scope = viewModelScope,
+        getCurrentId = { currentItem.id() },
+        onScoreChanged = {
+            _notifyResourceScoresChanged.value = true
+        }
+    )
+
+    private var currentPos = 0
+    private var sortByScores = false
+    private var selectingEnabled: Boolean = false
+
+    private val messageFlow: MutableSharedFlow<Message> = MutableSharedFlow()
+    private val currentItem: GalleryPresenter.GalleryItem
+        get() = galleryItems[currentPos]
+
+    private val _notifyResourceScoresChanged: MutableStateFlow<Boolean> =
+        MutableStateFlow(false)
+    val notifyResourceScoresChanged: StateFlow<Boolean> =
+        _notifyResourceScoresChanged
+
+    private val _setControlsVisibility: MutableStateFlow<Boolean> =
+        MutableStateFlow(false)
+    val setControlsVisibility: StateFlow<Boolean> =
+        _setControlsVisibility
+
+    private val _onNavigateBack: MutableStateFlow<Boolean> = MutableStateFlow(false)
+    val onNavigateBack: StateFlow<Boolean> = _onNavigateBack
+
+    private val _deleteResource: MutableStateFlow<Int?> = MutableStateFlow(null)
+    val deleteResource: StateFlow<Int?> = _deleteResource
+
+    private val _toastIndexFailedPath: MutableStateFlow<Path?> =
+        MutableStateFlow(null)
+    val toastIndexFailedPath: StateFlow<Path?> = _toastIndexFailedPath
+
+    private val _showInfoAlert: MutableStateFlow<ShowInfoData?> =
+        MutableStateFlow(null)
+    val showInfoAlert: StateFlow<ShowInfoData?> = _showInfoAlert
```

Comment:
I think this is hard to read and edit, let's redo it with [mvi-orbit](https://github.com/orbit-mvi/orbit-mvi)
We need to declare a state with fields like `controlsVisible`, `currentItem`, `selectingEnabled`. And side effects such as `NavigateBack`, `ShowInfoAlert`

Reply:
Great. I just applied and the viewModel is a bit cleaner. Thank man